File: iosdemo/ViewController.m
//
//  ViewController.m
//  iosdemo
//
//  Created by supergoodd on 4/23/15.
//  Copyright (c) 2015 Appodeal. All rights reserved.
//

#import "ViewController.h"
#import <AppodealAds/Appodeal.h>

@interface ViewController () <AODInterstitialDelegate, AODVideoAdDelegate, AODAdBannerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // set interstitial delegate
    [Appodeal setInterstitialDelegate:self];
    
   
    // set video ad delegate
    [Appodeal setVideoAdDelegate:self];
}

///////////////////////////////////////////////////////////////////

#pragma - mark Actions

- (IBAction)showInterstitialBanner:(id)sender {
    // show full screen banner
    [Appodeal show:self adType:INTERSTITIAL];
}

- (IBAction)showVideoAd:(id)sender {
    // show video ads
    [Appodeal show:self adType:VIDEO];
}

- (IBAction)showSmallBanner:(id)sender {
    // show small banner and set small banner delegate
    [Appodeal setBannerDelegate:self];
    [Appodeal show:self adType:BANNER_BOTTOM];
    // view banner from 1 adnetwork
    //[Appodeal showWithAdNetworkName:@"applovin" adType:BANNER_BOTTOM rootController:self];
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

///////////////////////////////////////////////////////////////////

#pragma mark - <AODInterstitialDelegate>

- (void)onInterstitialAdLoaded:(NSString*)adName isPrecache:(BOOL)isPrecache {
    NSLog(@"interstitial from %@ did load", adName);
}

- (void)onInterstitialAdFailedToLoad:(NSString*)adName {
    NSLog(@"interstitial from %@ failed to load", adName);
}

- (void)onInterstitialAdShown:(NSString*)adName {
    NSLog(@"interstitial from %@ failed to load", adName);
}

- (void)onInterstitialAdClicked:(NSString*)adName {
    NSLog(@"interstitial from %@ has been clicked", adName);
}

- (void)onInterstitialAdClosed:(NSString*)adName {
    NSLog(@"interstitial from %@ has been closed or dismissed", adName);
}

///////////////////////////////////////////////////////////////////

#pragma mark - <AODAdBannerDelegate>

- (void)onAdBannerLoaded:(UIView*)bannerView {
    NSLog(@"AdBanner loaded");
}

- (void)onAdBannerFailedToLoad {
    NSLog(@"AdBanner did failed to load");
}

- (void)onAdBannerShown {
    NSLog(@"AdBanner has been shown");
}

- (void)onAdBannerClicked {
    NSLog(@"onAdBanner has been clicked");
}

///////////////////////////////////////////////////////////////////

#pragma mark - <AODVideoAdDelegate>

- (void)onVideoLoaded:(NSString*)adName {
    NSLog(@"video ad from %@ did load", adName);
}

- (void)onVideoFailedToLoad:(NSString*)adName {
    NSLog(@"video ad from %@ failed to load", adName);
}

- (void)onVideoShown:(NSString*)adName {
    NSLog(@"video ad from %@ failed to load", adName);
}

- (void)onVideoClicked:(NSString*)adName {
    NSLog(@"video ad from %@ has been clicked", adName);
}

- (void)onVideoClosed:(NSString*)adName {
    NSLog(@"video ad from %@ has been closed or dismissed", adName);
}

- (void)onVideoFinished:(NSString*)adName {
    NSLog(@"video ad from %@ has been completed and user should rewarded virtual currency", adName);
}

@end

File: ios-swift-demo/appodeal-ios-swift-demo/ViewController.swift
//
//  ViewController.swift
//  appodeal-ios-swift-demo
//
//  Created by Andrey Tyshlaev on 03/06/15.
//  Copyright (c) 2015 Appodeal. All rights reserved.
//

import UIKit
import AppodealAds

class ViewController: UIViewController, AODInterstitialDelegate, AODAdBannerDelegate, AODVideoAdDelegate {
   
    @IBAction func showInterstitialBanner(sender: UIButton) {
        Appodeal.show(self, adType: AODAppodealAdType.INTERSTITIAL)
    }
    
    @IBAction func showVideoAd(sender: UIButton) {
        Appodeal.show(self, adType: AODAppodealAdType.VIDEO)
    }
    
    @IBAction func showSmallBanner(sender: UIButton) {
        Appodeal.show(self, adType: AODAppodealAdType.BANNER_BOTTOM)
    }

    
    override func viewDidLoad() {
        super.viewDidLoad()
        Appodeal.setInterstitialDelegate(self)
        Appodeal.setVideoAdDelegate(self)
        Appodeal.setBannerDelegate(self)
    }
    
    //MARK: - AODInterstitialDelegate
    
    func onInterstitialAdLoaded(adName: String!, isPrecache: Bool) {
        NSLog("interstitial from %@ did load", adName)
    }
    
    func onInterstitialAdFailedToLoad(adName: String!) {
        NSLog("interstitial from %@ failed to load", adName)
    }
    
    func onInterstitialAdShown(adName: String!) {
        NSLog("interstitial from %@ failed to load", adName)
    }
    
    func onInterstitialAdClicked(adName: String!) {
        NSLog("interstitial from %@ has been clicked", adName)
    }
    
    func onInterstitialAdClosed(adName: String!) {
        NSLog("interstitial from %@ has been closed or dismissed", adName)
    }
    
    //MARK: - AODAdViewDelegate
    
    func viewControllerForPresentingModalView() -> UIViewController! {
        return self;
    }
    
    func onAdBannerLoaded(bannerView: UIView!) {
         NSLog("AdBanner did load")
    }
    
    func onAdBannerFailedToLoad() {
        NSLog("AdBanner did failed to load")
    }
    
    func onAdBannerClicked() {
        NSLog("AdBanner has been clicked")
    }
    
    func onAdBannerShown() {
        NSLog("AdBanner has been shown")
    }
       
    //MARK: - AODVideoAdDelegate
    
    func onVideoLoaded(adName: String!) {
        NSLog("video ad from %@ did load", adName);
    }
    
    func onVideoFailedToLoad(adName: String!) {
        NSLog("video ad from %@ failed to load", adName);
    }
    
    func onVideoShown(adName: String!) {
        NSLog("video ad from %@ shown", adName);
    }
    
    func onVideoClicked(adName: String!) {
        NSLog("video ad from %@ has been clicked", adName);
    }
    
    func onVideoClosed(adName: String!) {
        NSLog("video ad from %@ has been closed or dismissed", adName);
    }
    
    func onVideoFinished(adName: String!) {
        NSLog("video ad from %@ has been completed and user should rewarded virtual currency", adName);
        
    }
}

